### PR TITLE
AI Minefield stuff

### PR DESCRIPTION
**Pull request by Lazejun**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
Currently All AI Faction placed Mines use Vanilla mines, nothing from the Templates, this PR changes this.
Found a few Typos in Templates and corrected those.

**fn_createAIcontrols.sqf**
Lowered amount of Mines from 60 to 45,
Made it use AP mines from Templates,
No changes to the rest of the File.
Added a Line logging Minefield Placement

**fn_minefieldAAF.sqf**
WavedCA places Mines upon Loosing the Attack on the Marker of the Attacking Side.
Side-Mine Selection is switched around so the Faction failing the Attack will place Mines on their Own Marker as Defensive Measure.

Tabs to Spaces,
Lowered amount of Mines from 60 to 30,
Added a Case for the Sides,
Made it use Template Mines, this one will be mixed AP and AT.
Updated Log Messages, it will only Log Creation of a Minefield if it successful, a Failure will only log if it Fails.
Changed Placement of Createmine from 100 to 50 ------> More density,
Increased the Radius between Multiple Minefields on the same Marker from 100 to 200 ------> More spacing between Minefields than before.

### Please specify which Issue this PR Resolves.
closes nothing

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
Use a Modset like RHS, as that has Different mines for each Faction, Vanilla uses the same all over.
Altis is pretty good to test.

For MinefieldAAF:
Calling mine Placement on a Marker:
```sqf
//Attacking Marker, Loosing Marker,
//Upon failing an Attack, the Attacker will place Defensive Mines on their Marker.
["Defender", "Attacker"] call A3A_fnc_minefieldAAF
```
Call an Attack onto a Marker and let the Attacker loose.
Then the Attacker will place Mines on their area.
```sqf
//Attacks
["Defender", "Attacker", 2] spawn A3A_fnc_wavedCA
```

For CreateAIControl Minefields:
As it needs to be near Frontlines, you need a Marker near the Control Marker.
```sqf
//This has a Control Marker nearby
[Teamplayer, "outpost_35"] remoteExec ["A3A_fnc_markerChange", 2];
//This is within spawn Range of the Control Marker
player setpos [3876.81,21507,0.00111389];
```

```sqf
//Reveal Mines for Playerside
{teamplayer revealMine _x} forEach allMines;
```
********************************************************
Notes:
Example how Mines will be Placed on a Control Marker

Example of Minefield Placement (8 Times on the same Outpost, that might rarely happen)

### CODE REVIEW

File: A3-Antistasi/functions/CREATE/fn_createAIcontrols.sqf [hunk 1]
```diff
@@ -158,10 +158,16 @@ else
...
-				_mineX = createMine ["APERSMine",_positionX,[],_size];
-				if (_sideX == Occupants) then {Occupants revealMine _mineX} else {Invaders revealMine _mineX};
+				if (_sideX == Occupants) then {
+				_mineX = createMine [ selectRandom (A3A_faction_occ getVariable "minefieldAPERS") ,_positionX,[],_size];
+   				Occupants revealMine _mineX;
+				} else {
+				_mineX = createMine [ selectRandom (A3A_faction_inv getVariable "minefieldAPERS") ,_positionX,[],_size];
+				Invaders revealMine _mineX;
+					};
 				};
```
Comment (CalebSerafin) [inline A3-Antistasi/functions/CREATE/fn_createAIcontrols.sqf:171]: We can move the if-statement outside the for-loop here.
```suggestion
				private _mines = ([A3A_faction_inv,A3A_faction_occ] select (_sideX == Occupants)) getVariable "minefieldAPERS";
				private _revealTo = [Invaders,Occupants] select (_sideX == Occupants);
				for "_i" from 1 to 45 do {
					_mineX = createMine [ selectRandom _mines ,_positionX,[],_size];
					_revealTo revealMine _mineX;
				};
```

File: A3-Antistasi/functions/CREATE/fn_minefieldAAF.sqf [hunk 2]
```diff
@@ -22,39 +20,49 @@ _distanceX = 500;
...
+for "_i" from 1 to 30 do
+{
+    if (sidesX getVariable [_markerX,sideUnknown] == Occupants) then {
+    _mineX = createMine [ selectRandom ((A3A_faction_inv getVariable "minefieldAT") + (A3A_faction_inv getVariable "minefieldAPERS")) ,_pos,[],50];
+    Invaders revealMine _mineX;
+    } else {
+    _mineX = createMine [ selectRandom ((A3A_faction_occ getVariable "minefieldAT") + (A3A_faction_occ getVariable "minefieldAPERS")) ,_pos,[],50];
+    Occupants revealMine _mineX;
+    };
+};
```
Comment (CalebSerafin) [inline A3-Antistasi/functions/CREATE/fn_minefieldAAF.sqf:65]: We can move the if-statement outside the for-loop here.
```suggestion
private _mines = ([A3A_faction_inv,A3A_faction_occ] select (sidesX getVariable [_markerX,sideUnknown] == Occupants)) getVariable "minefieldAT";
private _revealTo = [Invaders,Occupants] select (sidesX getVariable [_markerX,sideUnknown] == Occupants);
for "_i" from 1 to 30 do {
	_mineX = createMine [ selectRandom _mines ,_pos,[],50];
	_revealTo revealMine _mineX;
};
```

**Review by CalebSerafin — CHANGES_REQUESTED** [review]
The code is good. A couple mine spawning loops can be modifies to select the mines and side before entering the loop.

**Review by CalebSerafin — APPROVED** [review]
Code looks solid. I couldn't of done it any better myself.

**Comment by Lazejun** [comment]
Updated to use _base for Side checking as that works for all Cases, MarkerX was incompatible with Rebell Side winning a WavedCA causing wrong Mines to be placed by the loosing Attacker.

------> Loosing Occupants will place their Mines on their Makers.
------> Loosing Invaders will place their Mines on their Makers.

**Pull request merged**